python/cudf/cudf/core/dtypes.py

```diff
@@ -462,8 +430,47 @@ def serialize(self) -> Tuple[dict, list]:
     def deserialize(cls, header: dict, frames: list):
         return cls(header["precision"], header["scale"])
 
+    def __eq__(self, other: Dtype) -> bool:
+        if other is self:
+            return True
+        elif not isinstance(other, self.__class__):
+            return False
+        return self.precision == other.precision and self.scale == other.scale
+
+    def __hash__(self):
+        return hash(self._typ)
 
-class Decimal64Dtype(_BaseDtype):
+
+class Decimal32Dtype(DecimalDtype):
+    """
+    Parameters
+    ----------
+    precision : int
+        The total number of digits in each value of this dtype
+    scale : int, optional
+        The scale of the Decimal32Dtype. See Notes below.
+
+    Notes
+    -----
+        When the scale is positive:
+            - numbers with fractional parts (e.g., 0.0042) can be represented
+            - the scale is the total number of digits to the right of the
+            decimal point
+        When the scale is negative:
+            - only multiples of powers of 10 (including 10**0) can be
+            represented (e.g., 1729, 4200, 1000000)
+            - the scale represents the number of trailing zeros in the value.
+        For example, 42 is representable with precision=2 and scale=0.
+        13.0051 is representable with precision=6 and scale=4,
+        and *not* representable with precision<6 or scale<4.
+    """
```

Comment:
AFAICT the docstrings for all the decimal types are all identical aside from the use of the class name in the `scale` parameter. If you rewrite it as "The scale of the dtype. See Notes below" then you can put the docstring in the parent class and only write it once.